app/src/common/utils/routingUtils.js

```diff
@@ -25,10 +30,30 @@ export const copyQuery = (query = {}, namespacesToCopy = []) =>
     return acc;
   }, {});
 
+const getFilterNameByKey = (key) => {
+  const filterPrefixes = [FILTER_PREFIX, PREDEFINED_FILTER_PREFIX];
+  const isFilterRegExp = new RegExp(filterPrefixes.join('|'));
+  if (isFilterRegExp.test(key)) {
+    return key.split('.').pop();
+  }
+  return key;
+};
+
+const isEqualKeys = (newKey, oldKey) =>
+  newKey === oldKey || getFilterNameByKey(newKey) === getFilterNameByKey(oldKey);
+
+const mergeKeys = (oldQuery = {}, paramsToMerge = {}) =>
+  Object.keys(oldQuery).reduce(
+    (query, oldKey) =>
+      query.some((newKey) => isEqualKeys(newKey, oldKey)) ? query : [...query, oldKey],
+    Object.keys(paramsToMerge),
+  );
+
```

Comment:
I am working on the same part of the code now and it will be changed as part of the global task on filters. The best thing is to re-assign task to me and hold PR.